kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kroxylicious/CertManager.java

```diff
@@ -42,7 +44,8 @@ public CertManager() throws IOException {
      */
     public void deploy() {
         LOGGER.info("Deploy cert manager in {} namespace", Constants.CERT_MANAGER_NAMESPACE);
-        if (kubeClient().getNamespace(Constants.CERT_MANAGER_NAMESPACE) != null) {
```

Comment:
(non blocking comment - happy for this to be deferred) I don't think checking for a namespace is a good way to determine if something is installed on the Cluster.

A better approach is to check for a something like the CRD owned by the operator.    For instance, Red Hat have their own Cert Manager operator, which IIRC installs to openshift-operators.

Reply:
That's what I'm talking about with @im-konge. How can I check that?

Reply:
`kubectl get crds -l app=strimzi` this is for example for Strimzi
maybe there is a better way

Reply:
in principle, the installation yaml indicates that is should be `app=cert-manager`

Reply:
why worry about labels?  The CRD names are stable.   If one CRD belong to an operator is installed, I think that's a good enough proxy for the operator itself is installed.

kubectl get crd kafkas.kafka.strimzi.io

Reply:
I have added some changes to detect CRDs for cert manager. I believe using the label is easier to find out if there is something installed for cert manager

Reply:
Yeah sure, checking just one CRD makes sense.
I was thinking about scenario when the CRDs will be there, but the Deployment of operator wasn't applied (so it's f.e. leftover from testing or whatever)

Reply:
How can we know if it is a leftover or a complete installation? IMO if something was installed by a third party we shouldn't remove it, but we cannot know if it was installed by Red Hat or by another system tests from other products inside AMQ Streams...

Reply:
I mean, you can use the label as I mentioned before
```
kubectl get deployment -l app=strimzi --all-namespaces
```
and see it is running (you can do it via fabric8)

But I don't know about any other option without using the label and filtering all the Deployments and stuff.

TBH if you think that it will never happen, you can always skip it and really just check the CRD. It was just scenario that came to my mind.

Reply:
I have done that for cert manager deployment checking. Now we are looking for `app=cert-manager` in any deployment in any namespace.